Case 21727: Possible Entity Script Ban incident (RC 80)

danteruiz opened the pull request:
https://highfidelity.manuscript.com/f/cases/21727/Possible-Entity-Script-Ban-incident-80-0

ZappoMan commented on libraries/script-engine/CMakeLists.txt:
```diff
@@ -17,6 +17,6 @@ if (NOT ANDROID)
 endif ()
 
-link_hifi_libraries(shared networking octree shaders gpu procedural graphics material-networking model-networking ktx recording avatars fbx hfm entities controllers animation audio physics image midi)
+link_hifi_libraries(shared networking octree shaders gpu procedural graphics material-networking model-networking ktx recording avatars fbx hfm entities controllers animation audio physics image midi ui qml)
```
hmmmm this is worrisome... 

This might be problematic with the Entity Script Server and the Agent... this means they will become dependent on UI and QML???

hifi-gustavo commented:
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18216/HighFidelity-Beta-PR15208-47b9332.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18216/HighFidelity-Quest-Beta-PR15208-47b9332.apk)

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18216/HighFidelity-Beta-PR15208-47b9332.dmg

hifi-gustavo commented:
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18221/HighFidelity-Beta-PR15208-0e82691.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18221/HighFidelity-Quest-Beta-PR15208-0e82691.apk)

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18221/HighFidelity-Beta-PR15208-0e82691.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18221/HighFidelity-Beta-Interface-PR15208-0e82691.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18221/HighFidelity-Beta-PR15208-0e82691.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18221/HighFidelity-Beta-Interface-PR15208-0e82691.exe

danteruiz commented:
These test are based on my recent changes...
I build the servers and ran them locally on my machine. I added an entity and added an entity sever script to it. The API Users is not available for entity sever scripts. What I saw in my logs `[CRITICAL] [hifi.scriptengine] [6296] [entity-script-server] [about:Entities 1] [UncaughtException signalHandlerException] Error: Can't find variable: Users in https://dantescalves.com/hifi-content/entity-scripts/kick.js:6
[03/17 18:51:15] [CRITICAL] [hifi.scriptengine] [6296] [entity-script-server] [Backtrace]
[03/17 18:51:15] [CRITICAL] [hifi.scriptengine] [6296] [entity-script-server]     kickRandomUser(0.017483999952673912) at https://dantescalves.com/hifi-content/entity-scripts/kick.js:6
[03/17 18:51:15] [CRITICAL] [hifi.scriptengine] [6296] [entity-script-server]     <global>() at -1
`

I also ran an ac scripts (To my understanding the ac script is run on an agent) and in my agent log file see that the script does call the Users.kick function and that the C++ just calls the non ui code path based on the debug statements that I added

danteruiz commented:
Test plan for script api
Client Entity Script - https://dantescalves.com/hifi-content/entity-scripts/kick.js
click on the entity and if you have right to kick the dialog should appear

Client Server Script - https://dantescalves.com/hifi-content/entity-scripts/kick-entity-server.js
Nothing should happen. If you look at the logs for that EntityScriptServer. you should see that the script constantly crashes 

AC - script (load from the domain settings page) - https://dantescalves.com/hifi-content/entity-scripts/agentKick.js
nothing should happen, no user should be kicked. Since the agent running the script has not rights. The agent logs should give an error that the user has no rights to kick.
